libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -1466,21 +1481,27 @@ std::pair<vec_blk_t, bool> PbftManager::comparePbftBlockScheduleWithDAGblocks_(s
     cert_sync_block_.transactions.push_back(trx);
   }
 
-  auto calculated_order_hash = calculateOrderHash(dag_blocks_order, non_finalized_transactions);
+  const auto calculated_order_hash = calculateOrderHash(dag_blocks_order, non_finalized_transactions);
   if (calculated_order_hash != pbft_block->getOrderHash()) {
-    LOG(log_er_) << "Order hash incorrect. Pbft block: " << pbft_block->getBlockHash()
+    LOG(log_er_) << "Order hash incorrect. Pbft block: " << proposal_block_hash
                  << ". Order hash: " << pbft_block->getOrderHash() << " . Calculated hash:" << calculated_order_hash
                  << ". Dag order: " << dag_blocks_order << ". Trx order: " << non_finalized_transactions;
-    dag_blocks_order.clear();
-    return std::make_pair(std::move(dag_blocks_order), false);
+    return {{}, false};
+  }
+
+  // Check reward votes
+  auto reward_votes = vote_mgr_->updateRewardVotes(proposal_period - 1);
+  if (!vote_mgr_->checkRewardVotes(pbft_block, reward_votes)) {
+    LOG(log_er_) << "Failed verifying reward votes for proposed PBFT block " << proposal_block_hash;
+    return {{}, false};
   }
```

Comment:
We should, node have to save previous period cert votes as many as it can, then the node will have more higher chance to vote on peers proposal PBFT blocks